internal/app/singularity/registry.go

```diff
@@ -134,6 +134,15 @@ func (l *Library) copyFromCache(hash, name, to string) error {
 		return errNotInCache
 	}
 
+	sylog.Debugf("Removing %s (if exists) before copying cached image", to)
```

Comment:
It's a bit odd to put a message saying we might or might not be doing something. Can you make it explicit only if we *actually* remove the file?